cwms-data-api/src/main/java/cwms/cda/data/dto/basin/Basin.java

```diff
@@ -0,0 +1,195 @@
+package cwms.cda.data.dto.basin;
+
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.data.dto.basinconnectivity.Stream;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.NamedPgJsonFormatter;
+import cwms.cda.formatters.json.PgJsonFormatter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+@FormattableWith(contentType = Formats.NAMED_PGJSON, formatter = NamedPgJsonFormatter.class, aliases = {Formats.DEFAULT, Formats.JSON})
+@FormattableWith(contentType = Formats.PGJSON, formatter = PgJsonFormatter.class)
+public final class Basin implements CwmsDTOBase {
+    private final String basinName;
+    private final String officeId;
+    private final Stream primaryStream;
+    private final Double sortOrder;
+    private final Double basinArea;
+    private final Double contributingArea;
+    private final String parentBasinId;
+    private final List<Stream> streams;
```

Comment:
I would remove the streams and sub-basins altogether here. 

The client will handle the business of getting sub-basins. I think by querying basins and looking it their parentBasinIds.

Reply:
Addressed changes